test/suites/rbac.sh

```diff
@@ -0,0 +1,125 @@
+test_rbac() {
+  ensure_import_testimage
+
+  rbac_address="$(cat "${TEST_DIR}/rbac.addr")"
+  rbac_url="http://${rbac_address}"
+  key=$(curl -s "${rbac_url}/auth/discharge/info" | jq .PublicKey)
+  lxc config set rbac.api.key "${key}"
+  lxc config set rbac.api.url "${rbac_url}"
+  lxc config set "rbac.agent.url" "https://example.com"
+  lxc config set "rbac.agent.private_key" "${key}"
+  lxc config set "rbac.agent.public_key" "${key}"
+  lxc config set "rbac.agent.username" agent
+
+  (
+  cat <<EOF
+user1
+pass1
+EOF
+  ) | lxc remote add rbac "https://$LXD_ADDR" --auth-type candid --accept-certificate
+
+  curl "${rbac_url}/set-perms?user=user1" -X POST -d '{"": ["admin"]}'
+  lxc list rbac:
+  lxc list rbac: --all-projects
+  lxc launch testimage rbac:c1
+  lxc network create rbac:n1 --type bridge
+  lxc storage create rbac:s1 dir
+  lxc project create rbac:p1
+  lxc project rm rbac:p1
+  lxc profile create empty
+  lxc storage volume create rbac:s1 v1
+
+  testimage_fingerprint="$(lxc image ls -f csv | grep -F 'testimage' | cut -d, -f2)"
+
+  curl "${rbac_url}/set-perms?user=user1" -X POST -d '{"default": ["view"]}'
+  lxc project show rbac:default > /dev/null
+  lxc config show rbac:c1 > /dev/null
+  lxc image show "rbac:${testimage_fingerprint}" > /dev/null
+  lxc network show rbac:n1 > /dev/null
+  lxc profile show rbac:default > /dev/null
+  lxc storage volume show rbac:s1 v1 > /dev/null
+  ! lxc project set rbac:default user.foo=bar || false # Does not have  "manage-projects"
+  ! lxc project create rbac:p2 || false # Does not have "admin"
+  ! lxc delete rbac:c1 || false # Does not have "manage-containers"
+  ! LXC_LOCAL='' lxc_remote exec rbac:c1 -- true || false # Does not have "operate-containers"
+  ! lxc image delete "rbac:${testimage_fingerprint}" || false # Does not have "manage-images"
+  ! lxc network delete rbac:n1 || false # Does not have "manage-networks"
+  ! lxc profile delete rbac:empty || false # Does not have "manage-profiles"
+  ! lxc storage volume delete rbac:s1 v1 || false # Does not have "manage-storage-volumes"
```

Comment:
I quite like those inline comments explaining why it is expected to fail, thanks!